pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -413,3 +398,23 @@ def teardown(self) -> None:
             self.lightning_module.cpu()
             # clean up memory
             torch.cuda.empty_cache()
+
+
+class _FakeQueue(list):
```

Comment:
afaik subclassing mutable builtins (especially dict and list) can sometimes lead to unexpected results. Maybe use [`UserList`](https://docs.python.org/3/library/collections.html#collections.UserList) as base class instead?

Reply:
@justusschock I don't think that's true nowadays: https://stackoverflow.com/questions/25464647/list-vs-userlist-and-dict-vs-userdict

`list` is the recommended approach for simple extension
and `collections.*` for complex extension

Reply:
Quoting Raymond Hettinger (Python core dev):

"FWIW, the UserDict class in Py3.x was relocated to the collections
module.  It isn't gone.  We made an effort to kill it but found that
there were compelling use cases that were not a cleanly solved by any
other approach."

https://bugs.python.org/issue5402